agent/eventhandler/task_handler_types.go

```diff
@@ -186,14 +186,16 @@ type setStatusSent func(event *sendableEvent)
 
 // setContainerChangeSent sets the event's container change object as sent
 func setContainerChangeSent(event *sendableEvent) {
-	if event.containerChange.Container != nil {
+	if event.containerChange.Container != nil &&
+		event.containerChange.Container.GetSentStatus() < event.containerChange.Status {
 		event.containerChange.Container.SetSentStatus(event.containerChange.Status)
 	}
 }
 
 // setTaskChangeSent sets the event's task change object as sent
 func setTaskChangeSent(event *sendableEvent) {
-	if event.taskChange.Task != nil {
+	if event.taskChange.Task != nil &&
```

Comment:
can you just assign `event.taskChange.Task` to a var and reuse it in this method so that it's easier to read this code?